client/src/Actions/userSlice.ts

```diff
@@ -0,0 +1,66 @@
+import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+export type UserType = {
+  id: number;
+  email: string;
+  password: string;
+  name: string;
+  surname: string;
+  avatar: string;
+};
+
+type UserState = {
+  user: UserType | null;
+  status: string | null;
```

Comment:
A może undefined zamiast null? Wtedy by było user?: UserType, ja preferuje taką składnie ale mogę się dopasować, luźna propozycja

Reply:
ok a jak mamy reject z requesta to wtedy musimy do usera przypisac wartość null, bo go nie ma i wtedy musi być type null. Chyba że masz jakiś pomysł jak to rozwiązć, bo ja nw ? :)

Reply:
Znaczy możemy przypisać undefined wtedy i bedzie to samo